low-level/files/opt/seagate/sspl/setup/sspl_test.py

```diff
@@ -56,14 +57,35 @@ def validate(self):
             "cortx-sspl-test": None
             }
         # python 3rd party package dependency
-        pip3_3ps_packages_test = {
-            "Flask": "1.1.1"
+        pip3_3ps_packages_dep = {
+            "Flask": "1.1.1",
+            "coverage": "5.5"
             }
+        # Collect installed python pkg from dependency list.
+        installed_pip3_pkgs = {}
+        for pkg in pkg_resources.working_set:
+            if pkg.project_name in pip3_3ps_packages_dep:
+                installed_pip3_pkgs[pkg.project_name] = pkg.version
+        # Validate pip3 python pkg with required version.
+        for pkg, version in pip3_3ps_packages_dep.items():
```

Comment:
why two loops, can be done in one loop

Reply:
In the first loop, I am getting all the pip pkgs installed on the node, and involving it in the list only if we are depending on it.  so we get installed_dep
In the second loop, I am checking if we have (dep - installed_dep) as non-zero then install those pkgs, also if the version number of installed_dep does not match then install the correct version.

How can we do it in a single loop?? I have no clue.
Another approach is still taking two loops.
In the first loop while retrieving the installed pkg we can check if that pkg is there in dep if it matches the version we can remove it from the dep list,  even then, we would need a second loop to go over pkg that is remaining in dep list.

Reply:
You are traversing through every installed package and then comparing if any of them matches with the packages you have in dependency list. Ideally it should be other way around.

Reply:
@jjxg, the working set is bigger so traversing through the installed pkg and then for each installed pkg checking if it is there in small set of dependency is better as compared to for each dependency pkg traverse big installed pkg list.

Reply:
addressed 👍